src/realloc-buffer.c

```diff
@@ -392,3 +409,30 @@ int realloc_buffer_printf(ReallocBuffer *b, const char *fmt, ...) {
         }
 
 }
+
+int realloc_buffer_memchr(ReallocBuffer *buffer, uint8_t c) {
+        const uint8_t *p, *q;
+        size_t l, d;
+        int ret;
+
+        /* Looks for byte 'c' in the buffer. Returns -ENXIO when we can't find the byte. Returns an index >= 0 if
+         * found. Returns -EOVERFLOW if found but the index doesn't fit in "int" */
+
+        l = realloc_buffer_size(buffer);
+        if (l <= 0)
+                return -ENXIO;
+
+        p = realloc_buffer_data(buffer);
+        q = memchr(p, c, l);
+        if (!q)
+                return -ENXIO;
+
+        d = (size_t) (q - p);
+        ret = (int) d;
+        if (ret < 0)
+                return -EOVERFLOW;
+        if ((size_t) ret != d)
+                return -EOVERFLOW;
```

Comment:
well, it's a paranoia check... all of this should have failed much earlier on...

But actually, it might make sense to add some size check to realloc_buffer_acquire() to explicitly never allow the thing to grow larger than 1G or so... Will prep another commit for that...